File: pairs/pair.js
/**
 * pairs - This function takes an array of elements and returns an array of all unique pairs of those elements.
 * Each pair is represented as an array containing two unique elements.
 * @param {*} elements
 * @returns
 */
const pairs = (elements) => {
  let result = [];
  for (let i = 0; i < elements.length; i++) {
    for (let j = i + 1; j < elements.length; j++) {
      result.push([elements[i], elements[j]]);
    }
  }
  return result;
};

console.log(pairs(["a", "b", "c", "d"]));
console.log(pairs([1, 2, 3, 4]));
console.log(pairs(["apple", "banana", "cherry"]));
console.log(pairs([true, false, true]));

File: pairs/pair.py
def pairs(elements):
    """_summary_

    Args:
        elements (_type_): _description_ an array of elements

    Returns:
        _type_: a list of pairs of elements
        The function takes an array of elements and returns a list of pairs of unique elements.
        Each pair is a list containing two elements from the input array.   
    Complexity:
    Time: O(n^2)
    Space: O(n^2)
    """
    res = []
    for i in range(0, len(elements)):
        for j in range(i+1, len(elements)):
            res.append([elements[i], elements[j]])
    return res


print(pairs(["a", "b", "c"]))
print(pairs([1, 2, 3, 4]))
print(pairs([1, 2, 3, 4, 5]))
print(pairs([]))
print(pairs(["a", "b", "c", "d", "e"]))
